### PR TITLE
Fix LINKS biocontainer

### PR DESCRIPTION
**Pull request by TomHarrop**
The biocontainers built from this package are not working for LINKS 2.x.

- 2.0.0--h7d875b9_0 fails because `perl` is missing (`bash: perl: command not found`)
- 2.0.1--h7d875b9_0 fails because LINKS doesn't get installed (`bash: LINKS_CPP: command not found`) 

I added `perl` as a runtime dependency to fix the former. I believe switching to `make install` fixes the latter. ~I can't test the container,~ @MurathanGoktas  could you take a look please?

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/links/build.sh [hunk 1]
```diff
@@ -1,16 +1,7 @@
...
-
-mkdir -p ${PREFIX}/bin/
-mkdir -p ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/bin
-mkdir -p ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/src
-
-cp src/LINKS_CPP ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/src
-cp bin/* ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/bin
-cp bin/LINKS* ${PREFIX}/bin/
-
+./configure && make && make install
```
Comment (MurathanGoktas) [inline recipes/links/build.sh:4]: Can you change this line to 
`./configure --prefix=${PREFIX}`
 `make`
 `make install`

File: recipes/links/build.sh [hunk 2]
```diff
@@ -1,16 +1,7 @@
...
 
 echo "#!/bin/bash" > ${PREFIX}/bin/LINKS-make
 echo "make -f $(command -v ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/bin/LINKS-make) \$@" >> ${PREFIX}/bin/LINKS-make
```
Comment (MurathanGoktas) [inline recipes/links/build.sh:7]: Can you delete lines 15 and 16?
`echo "#!/bin/bash" > ${PREFIX}/bin/LINKS-make`
`echo "make -f $(command -v ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/bin/LINKS-make) \$@" >> ${PREFIX}/bin/LINKS-make`

**Comment by TomHarrop** [comment]
Thanks for the help @MurathanGoktas , I have made those changes.

**Comment by MurathanGoktas** [comment]
Thanks for fixing @TomHarrop I didn't realize this was broken.

**Comment by TomHarrop** [comment]
I think it's [failing](https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=11842&view=logs&j=e14e69ff-a0ae-55c4-b71d-229b239cfb2f&t=7df82132-b284-504b-53d6-7d3e63519572&l=413) because the full path `/usr/bin/make` isn't where conda installs `make`.

https://github.com/bcgsc/LINKS/blob/bd01756ca6157ccb96fbb07fc49be7ce06d92a06/bin/LINKS-make#L1

I guess the shebang could be changed to `/usr/bin/env make` but I don't know if that will work with conda.

File: recipes/links/build.sh [hunk 3]
```diff
@@ -1,16 +1,6 @@
...
-mkdir -p ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/bin
-mkdir -p ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/src
-
-cp src/LINKS_CPP ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/src
-cp bin/* ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/bin
-cp bin/LINKS* ${PREFIX}/bin/
-
-
-echo "#!/bin/bash" > ${PREFIX}/bin/LINKS-make
-echo "make -f $(command -v ${PREFIX}/bin/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/bin/LINKS-make) \$@" >> ${PREFIX}/bin/LINKS-make
```
Comment (MurathanGoktas) [inline recipes/links/build.sh:16]: Yes looks like these two lines were dealing with the conda make the problem. So let's try

```
mv ${PREFIX}/bin/LINKS-make ${PREFIX}/bin/LINKS-make-real
echo "#!/bin/bash" > ${PREFIX}/bin/LINKS-make
echo "make -f $(command -v ${PREFIX}/bin/LINKS-make-real) \$@" >> ${PREFIX}/bin/LINKS-make
```

**Review by MurathanGoktas — APPROVED** [review]
@TomHarrop thanks for the work. Looks good to me.

**Comment by TomHarrop** [comment]
I just tested LINKS in the container from the [LinuxArtifacts URL](https://artprodsu6weu.artifacts.visualstudio.com/A27bcc0f9-b22c-4418-83d8-514809b37fb2/25074ded-3133-43ca-af80-895ab87b53cb/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2Jpb2NvbmRhL3Byb2plY3RJZC8yNTA3NGRlZC0zMTMzLTQzY2EtYWY4MC04OTVhYjg3YjUzY2IvYnVpbGRJZC8xMTg1NC9hcnRpZmFjdE5hbWUvTGludXhBcnRpZmFjdHM1/content?format=zip) and it works as expected, so I would say this is fixed. Thanks again for helping with this.

**Comment by TomHarrop** [comment]
@BiocondaBot please add label

**Review by dpryan79 — APPROVED** [review]

**Pull request merged**